prestoadmin/yarn_slider/server.py

```diff
@@ -248,3 +250,79 @@ def mk_data_dir():
              'user': app_config.get_user(),
              'group': app_config.get_group()})
     return sudo(command)
+
+
+def get_slider_status(conf):
+    status_cmd = '%s status %s' % (get_slider_bin(conf), conf[APP_INST_NAME])
+    with hide('running', 'stdout', 'stderr'):
+        status_json = degarbage_json(run_slider(status_cmd, conf))
+    return json.loads(status_json)
+
+
+ROLENAME_COORDINATOR = 'COORDINATOR'
+ROLENAME_WORKER = 'WORKER'
+
+
+def get_instances(slider_status, rolename):
+    return slider_status['instances'][rolename]
+
+
+def identity(x):
+    return x
+
+
+def get_coordinator(slider_status, filtr=identity):
+    coordinators = slider_status['status']['live'][ROLENAME_COORDINATOR]
+    return [filtr(v) for v in coordinators.values()]
+
+
+def get_workers(slider_status, filtr=identity):
+    try:
+        workers = slider_status['status']['live'][ROLENAME_WORKER]
+    except KeyError:
+        workers = {}
+
+    return [filtr(v) for v in workers.values()]
+
+
+def get_status_transformer():
+    app_config = AppConfigJson()
+
+    def transform_status(container):
+        host = container['host']
+        node_uri = 'http://%s:%s' % (
+            container['host'], app_config.get_presto_server_port())
+        is_running = container['state']
+        role = container['role'].lower()
+
+        return host, is_running, role, node_uri
+    return transform_status
+
+
+@task
+@requires_config(SliderConfig)
+@task_by_rolename(SLIDER_MASTER)
+def status():
+    """
+    Display status information about the cluster.
+    """
+    conf = env.conf
+    slider_status = get_slider_status(conf)
+    coordinators = None
+
+    try:
+        coordinators = get_coordinator(slider_status)
+    except KeyError:
+        abort(
+            'No coordinator found. This may be because presto is still '
+            'starting. Check the application status')
```

Comment:
I've only seen this happen in the short period between starting the app instance and it coming up (which is why there are a bunch of places where we wait for presto to come up and go down in the new product tests).

How does the following sound:
1. Add information about the workers, if available.
2. Change the error to tell the user to try wait a minute and try again or check the status via the ResourceManager and give the URL.
3. Possibly add another command to dump the raw slider JSON for debugging?